### PR TITLE
Fixes pretty.names argument for several plotting functions. Closes #1523

### LINKED ISSUE #1523: plotBMRBoxplots fails when the same learner is used multiple times (with different ids)
```
lrns = list(
  rf = makeLearner("classif.randomForest", id = "rf1"),
  rf2 = makeLearner("classif.randomForest", id = "rf2")
)


bm = benchmark(lrns, iris.task, hout)

plotBMRBoxplots(bm)
```

bug

This is because the `shortname` is the same for both learners, when I set `pretty.names=FALSE` it works

### PR DESCRIPTION
**Pull request by florianfendt**
Output of plots were confusing when pretty.names was on and learners had the same id.
Now we stop with an error message which identifies the (first) problematic learner.

Affected functions:
* plotResiduals
* plotBMRBoxplots
* plotBMRSummary
* plotBMRRanksAsBarCharts

Connected issue: #1523

### CODE REVIEW

File: R/helpers.R [hunk 1]
```diff
@@ -85,3 +85,15 @@ replaceDupeMeasureNames = function(measures, x = "id") {
...
 }
+
+# checks for duplicated entries in learner.names and stops
+# with error message containing the learner name that appeared more
+# than once
+checkDuplicatedLearnerNames = function(learner.names) {
+  dupl = duplicated(learner.names)
+  if (any(dupl)) {
+    dupl.learner = learner.names[which.first(dupl)]
+    stop(sprintf("Learner short names are not unique for: %s ",
```
Comment (ja-thomas) [inline R/helpers.R]: `stopf()`

Comment (ja-thomas) [inline R/helpers.R]: Also add a sentence "set `pretty.names=FALSE` to solve this" or something like that

File: R/helpers.R [hunk 2]
```diff
@@ -85,3 +85,15 @@ replaceDupeMeasureNames = function(measures, x = "id") {
...
   unlist(meas.names)
 }
+
+# checks for duplicated entries in learner.names and stops
+# with error message containing the learner name that appeared more
+# than once
+checkDuplicatedLearnerNames = function(learner.names) {
+  dupl = duplicated(learner.names)
+  if (any(dupl)) {
+    dupl.learner = learner.names[which.first(dupl)]
```
Comment (jakob-r) [inline R/helpers.R]: what if multiple names are duplicated? the error will just return one of the pairs. what about `unique(learner.names[dupl]))`.

File: R/helpers.R [hunk 3]
```diff
@@ -85,3 +85,15 @@ replaceDupeMeasureNames = function(measures, x = "id") {
...
+
+# checks for duplicated entries in learner.names and stops
+# with error message containing the learner name that appeared more
+# than once
+checkDuplicatedLearnerNames = function(learner.names) {
+  dupl = duplicated(learner.names)
+  if (any(dupl)) {
+    dupl.learners = unique(learner.names[dupl])
+    stopf("Learner short names are not unique for: %s. \n  Set 'pretty.names = FALSE' to resolve this.",
+      collapse(dupl.learners))
```
Comment (larskotthoff) [inline R/helpers.R]: The error message is very specific. I would either make it more generic, or move this function to a different file (plotUtils.R?).

**Comment by florianfendt** [comment]
@larskotthoff 
I put the checkDuplicated... function into a new file now: utils_plot.R

**Pull request merged**